Updated/refactored/tested config file JSON schema

edonv opened the pull request:
Made some thorough updates to `config.schema` file based on latest `dev` branch version. Started discussing in #520.

I'm also going to open a PR with [JSON Schema Store](https://www.schemastore.org/json/) to include this file in the JSON Schema Store for use with their text editor extensions, then eventually with a JSON Editor to make a web-based config editor with smart suggestions and whatnot.

The way it will work with JSON Schema Store is that the schema file won't live in their repo, but instead will point to the GitHub-hosted `config.schema` in the `ChordPro/chordpro` repo on the `master` branch.

sciurius commented:
I use json-schema-validator-2.2.6-lib.jar (https://mvnrepository.com/artifact/com.github.fge/json-schema-validator/2.2.6) for json validation, and it claims that the schema is invalid.

edonv commented:
While I'll say that it does seem slightly out-of-date, I'd be happy to get to the bottom of it. Does it give any debugging details? I used https://www.jsonschemavalidator.net/ to manually confirm that all of the sample files I have (the config files that come packaged with the program PLUS my own config files) all successfully work with my schema.

edonv commented:
## Errors

Ok seems like the only errors are two instances of empty `required` fields. That's not something I usually do, and I didn't even do it consistently in this schema. I started to add it to make another validator happy by specifying that none of the properties are required, but then I saw in the JSON Schema site docs that it's never supposed to be an empty array. I stopped adding that, but I guess I forgot to go back and remove the ones I had added. **I've now deleted them.**

## Warnings

The warnings all seem to just be about unrecognized keywords that are all just keywords added in later versions of JSON Schema:

- `$comment`
- `$id`
- `examples`
- `const`

The only other warning is that it doesn't like that the `fontconfig` has a hash key of an empty string `""`, but that's just what it is.

Should be good on this next commit.
